### PR TITLE
Adds collections integrity unit tests

### PR DESCRIPTION
**Pull request by ggirotto**
Also adds updated - added `resources` and made other small adjustments - collections to `collections/` folder.

### CODE REVIEW

File: firebase/test/collections/integrity-check.spec.ts [hunk 1]
```diff
@@ -1,6 +1,61 @@
...
+
+  before(async () => {
+    localCollectionsPaths = await fs.promises.readdir(`${process.cwd()}/collections`);
+
+    const fsGateway = Provider.instance.fileSystemGateway;
+    const localCollections = await fsGateway.readDirFilesAsStrings("./collections");
+    rawLocalCollections = localCollections.map((collection) => JSON.parse(collection));
+  });
+
+  it("should have at least one collection", () => assert.ok(rawLocalCollections.length > 0));
```
Comment (matuella) [inline firebase/test/collections/integrity-check.spec.ts:17]: Why should it? I Don't think it's an useful test

File: firebase/test/collections/integrity-check.spec.ts [hunk 2]
```diff
@@ -1,6 +1,61 @@
...
+  it("should have unique ids amongst themselves", () => {
+    const collectionIds: string[] = [];
+    rawLocalCollections.forEach((collection) => {
+      const collectionId: string = collection["id"];
+      assert.ok(!collectionIds.includes(collectionId), `Duplicate collection id "${collectionId}"`);
+      collectionIds.push(collectionId);
+    });
+  });
+
+  it("should have unique memo ids amongst themselves", () => {
```
Comment (matuella) [inline firebase/test/collections/integrity-check.spec.ts:39]:
```suggestion
  it("should have unique memo ids in each collection", () => {
```

Why should they have a unique global id?

File: firebase/test/collections/integrity-check.spec.ts [hunk 3]
```diff
@@ -1,6 +1,61 @@
...
+          !memosIds.includes(memoUniqueId),
+          `Duplicate memo id "${memoUniqueId}" in collection "${collection}"`
+        );
+        memosIds.push(memoUniqueId);
+      });
+    }
+  });
+
+  it("should have expected JSON structure", () => {
+    const schemaValidator = Provider.instance.schemaValidator;
```
Comment (matuella) [inline firebase/test/collections/integrity-check.spec.ts:55]: it seems odd to use provider here, why not create an instance of the expected schema validator?

**Review by matuella — COMMENTED** [review]
Shouldn't we move these tests to integration tests of some sort?

**Comment by matuella** [comment]
Also, I've already removed the test collection in `https://github.com/olmps/memo/pull/276`. Will merge both #275 and #276 so you can rebase on that.

File: firebase/test/collections/integrity-check.spec.ts (hunk 2 above)
Comment (ggirotto) [reply firebase/test/collections/integrity-check.spec.ts:39]: I copied this test from the client, I though it was a definition, but we may remove this test

**Review by matuella — COMMENTED** [review]
Will these collections properly run at the same time that the workflow is merged into `main`?

Like, we're developing in `server`, then, when merging into `main`, will the `.github/` changes work effectively with the script itself?

**Comment by ggirotto** [comment]
> Will these collections properly run at the same time that the workflow is merged into `main`?
> 
> Like, we're developing in `server`, then, when merging into `main`, will the `.github/` changes work effectively with the script itself?

Good catch. Updated `update-collections` workflow to run the integrity tests before running the collections sync.

**Comment by matuella** [comment]
> Good catch. Updated update-collections workflow to run the integrity tests before running the collections sync.

But does it change the fact that all collections are going to be merged before the workflow is available?

**Comment by ggirotto** [comment]
> But does it change the fact that all collections are going to be merged before the workflow is available?

I didn't understood your question. The `update-collections` is the one who merges the collections in Firestore. If it fails in the tests lane, it doesn't continue with the merging process.

**Comment by matuella** [comment]
This workflow will only work when changes are pushed to `main` branch. All collections here in this pull will be available at the same time as the workflow in the `main` branch, which, in my naive perspective, wouldn't run the sync, as it will only detect such changes, in that particular folder, after it has been available in `main`.

TLDR: shouldn't we split merging the collections after the server is available in main?

**Comment by ggirotto** [comment]
Actually I believe they'll be synched, because the workflow runs on every push on main detecting the changes since the last merge commit, which includes all added collections. It'll compare the merge commit in `main` with [b15d97](https://github.com/olmps/memo/commit/b15d976f7b7830cee7cac48623ab4dbf096b910c), which is the last commit, and in this difference these added collections will be considered.

**Review by matuella — APPROVED** [review]
Then I believe we should test it and see if it working as expected, otherwise split the `server` merge into the actual changes and the collections to-be-added.

I'll aprove this nonetheless, but please keep this is mind.
